### DISCUSSION TITLE
Interactive R code examples

### DISCUSSION THREAD

**Discussion by flaming-codes** [flaming-codes/crane-app#56] [General]
# About

It would be nice to have either interactive code examples inside the app directly (like Codesandbox) or hosted on another service like [DataCamp](https://www.datacamp.com/workspace/sign_up?redirect=https%3A%2F%2Fapp.datacamp.com%2Fworkspace%2Fworkspaces).

# Considerations

- at best no costs associated
- dependency has to be installable in the env
- code might only run for older R-version, so it has be selectable

Maybe using Jupyter notebooks is a solution.

**Comment by lschoenm** [comment]
While definitely a good idea, there are some challenges: 
- Many functions are designed for datasets (sometimes of a particular format/with certain data types)
- Some functions might only to be used in a workflow, which is difficult to simulate in short code snippets

Nevertheless, package creators are encouraged to provide examples for the functions of their packages, and those examples are checked and automatically run before each submission to CRAN. This means that theoretically it should be possible to use those on our site. 

Unfortunately I think on the website they are only shown as automatically compiled PDFs, but maybe there is a way to get them through other means than scraping.
